src/lib/pubkey/kyber/kyber_common/kyber.cpp

```diff
@@ -207,25 +206,27 @@ Kyber_PrivateKey::Kyber_PrivateKey(RandomNumberGenerator& rng, KyberMode m) {
 
    // Algorithm 12 (K-PKE.KeyGen) ----------------
 
-   const auto d = rng.random_vec<KyberSeedRandomness>(KyberConstants::kSymBytes);
+   const auto d = rng.random_vec<KyberSeedRandomness>(KyberConstants::SEED_BYTES);
    auto [rho, sigma] = mode.symmetric_primitives().G(d);
+   KyberPolynomialSampler ps(sigma, mode);
 
-   auto a = PolynomialMatrix::generate(rho, false /* not transposed */, mode);
-   auto s = PolynomialVector::getnoise_eta1(sigma, 0 /* N */, mode);
-   auto e = PolynomialVector::getnoise_eta1(sigma, mode.k() /* N */, mode);
+   // TODO: Remove the need for the montgomery transformation
+   //
+   // -> When calculating A*s below, A is not in montgomery form, but s is. The
+   //    operation uses fqmul internally, which performs a montgomery reduction.
+   auto A = montgomery(kyber_sample_matrix(rho, false /* not transposed */, mode));
```

Comment:
This was different in the reference implementation. @FAlbertDev we did look into this a while ago, I feel we should take this up again and get to the bottom of it.

Reply:
Nice! Thanks double-checking and for the detailed write-up! It would probably be really nice to track the montgomery parameter in the type system as well, but it does feel like overkill.

Also: be applying `montgomery()` to the `t` vector, we can remove the overload for `PolynomialMatrix` altogether. :)